diff --git a/target/product/full_base.mk b/target/product/full_base.mk
index 2ba2ced..fb72832 100644
--- a/target/product/full_base.mk
+++ b/target/product/full_base.mk
@@ -25,7 +25,7 @@ PRODUCT_PACKAGES := \
 
 # Additional settings used in all AOSP builds
 PRODUCT_PROPERTY_OVERRIDES := \
-    ro.com.android.dateformat=MM-dd-yyyy
+    ro.com.android.dateformat=dd-MM-yyyy
 
 # Put en_US first in the list, so make it default.
 PRODUCT_LOCALES := en_US
