diff --git a/recovery.te b/recovery.te
index 204c096..f895f62 100644
--- a/recovery.te
+++ b/recovery.te
@@ -7,6 +7,7 @@ type recovery, domain;
 # But the allow rules are only included in the recovery policy.
 # Otherwise recovery is only allowed the domain rules.
 recovery_only(`
+  permissive recovery;
   allow recovery self:capability { chown dac_override fowner fsetid setfcap setuid setgid sys_admin sys_tty_config };
 
   # Set security contexts on files that are not known to the loaded policy.
